src/restLevelClient/generateClient.ts

```diff
@@ -153,7 +165,10 @@ function getClientFactoryBody(
     declarations: [{ name: "baseUrl", initializer: baseUrl }]
   };
 
-  const { credentialScopes, credentialKeyHeaderName } = getAutorestOptions();
+  const {
```

Comment:
Here as well check if getAutorestOptions have credentialScopes or credentialKeyHeaderName and use it, otherwise try to get security info from `getSecurityInfoFromModel`